File: src/main/java/j/examples/rules/feedback/PdfExtensionFileRule.java
package j.examples.rules.feedback;

import java.io.File;
import java.util.ArrayList;
import java.util.List;

public class PdfExtensionFileRule {

    public List<String> validate(File file) {
        var messages = new ArrayList<String>();

        String fileExtension = file.getName().substring(file.getName().lastIndexOf('.') +1);

        if (!fileExtension.equals("pdf")) {
            messages.add("The file %s is not a .pdf file".formatted(file));
        }

        return messages;
    }
}

File: src/main/kotlin/k/examples/rules/feedback/PdfExtensionFileRule.kt
package k.examples.rules.feedback

import java.io.File

class PdfExtensionFileRule {
    fun validate(file: File): List<String> {
        val messages = mutableListOf<String>()
        if (file.extension != "pdf") {
            messages += "The file ${file} is not a .pdf file"
        }
        return messages
    }
}